common/lib/src/realm_common_base.dart

```diff
@@ -16,12 +16,36 @@
 //
 // //////////////////////////////////////////////////////////////////////////////
 
+/// An enum controlling the base type for a [RealmModel].
+///
+/// {@category Annotations}
+enum ObjectType {
+  /// A standalone top-level object that can be persisted in Realm. It can link
+  /// to other objects or collections of other objects.
+  topLevel('RealmObject'),
```

Comment:
Yeah, I was debating that for a while as well - I thought things are getting a little confusing as everything is a realm object, so it may not be clear to a user what does it mean to have a `RealmModel(ObjectType.realmObject)`. Additionally, it reads very repetitive - `ObjectType.embeddedObject` vs `ObjectType.embedded`. But I also agree `topLevel` is not super clear either - I don't have strong opinions as none of the options seems much better than the others.

Reply:
we agreed to rename this to `realmObject`. Resolving